### PR TITLE
Exclude Scarf Usage Data Collection in CI Environments

### PR DESCRIPTION
**Pull request by kaxil**
Most of the the CI systems add "CI=true" env var.

Refereces:

- https://docs.pytest.org/en/stable/explanation/ci.html
- https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#default-environment-variables
- https://docs.travis-ci.com/user/environment-variables/
- https://docs.gitlab.com/ee/ci/variables/predefined_variables.html
- https://circleci.com/docs/variables/#built-in-environment-variables
- https://www.jenkins.io/doc/book/pipeline/jenkinsfile/#using-environment-variables
- https://adamj.eu/tech/2020/03/09/detect-if-your-tests-are-running-on-ci/
- https://github.com/The-Compiler/pytest-vw/blob/master/pytest_vw.py

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by ashb — APPROVED** [review]

File: airflow/utils/usage_data_collection.py [hunk 1]
```diff
@@ -70,6 +75,26 @@ def _version_is_prerelease(version: str) -> bool:
     return parse(version).is_prerelease
 
 
+def _is_ci_environ() -> bool:
+    """Return True if running in any known CI environment."""
+    if os.getenv("CI") == "true":
```
Comment (jedcunningham) [inline airflow/utils/usage_data_collection.py:80]: Should we be flexible for case here too? Or maybe just bail if the `CI` env var is set at all?

Comment (kaxil) [reply airflow/utils/usage_data_collection.py:80]: Based on the convention of what these CI system use, `CI` (caps) seems to be standard so 🤷

Comment (jedcunningham) [reply airflow/utils/usage_data_collection.py:80]: Sorry, I mean the casing of the `true`.

Comment (kaxil) [reply airflow/utils/usage_data_collection.py:80]: ah, don't think that is worth it too tbh, all (that have `CI` set) of them have `CI=true`

**Pull request merged**